inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -90,6 +86,17 @@ std::string name(const Node& n) {
     return n->get_friendly_name();
 }
 
+template <typename Tensor>
+std::string tensor_names(const Tensor& t) {
```

Comment:
Why do you have a template parameter `Tensor`? If I remember right we always work with `ngraph::description::Tensor`.

Reply:
It's template for single compilation unit and I was to lazy to check accurate type of Tensor because it doesn't really matter here. 
From C++14 I may be put inplace generic lambda expression and call it to create names (`[](const auto& t){/*...*/}`).

Primary goal of this change is to split check expression from error message creation. 

IMHO, `compare_function` should contains code which will be easy to read. In this situation, that's mean, it will be easy to find what is check. In the next step how is check and message creation - doesn't care.  It doesn't mean that crated message can be worthless. It has to bring useful information but creation of message should take as small as possible number lines and it shouldn't obscure whole algo.

Reply:
I understand that you made this change to separate the place with error message creation.

But I prefer to have `ngraph::description::Tensor` as an argument of this method instead of template argument. I understand that this code doesn't affect the binary size or anybody else, but if I see the function with template arguments first of all I tried to understand where we have a variability for this parameter.

IMHO at this current case I prefer to avoid template parameter.
Moreover your example with lambda with auto parameter is a little bit different thing, because in the majority of situations we use local lambda functions, it means that it is easier to find lambda usages and understand what happens.

Reply:
Done